server/src/test/java/org/apache/druid/metadata/SQLMetadataConnectorTest.java

```diff
@@ -305,6 +303,140 @@ public void testIsTransientException()
     );
   }
 
+  @Test
+  public void test_useShortIndexNames_true_tableIndices_areNotAdded_ifExist()
+  {
+    tablesConfig = new MetadataStorageTablesConfig(
+        "druidTest_base_table_name_true",
+        null, null, null, null, null, null, null, null, null, null, null,
+        true
+    );
+    connector = new TestDerbyConnector(new MetadataStorageConnectorConfig(), tablesConfig);
+
+    final String segmentsTable = tablesConfig.getSegmentsTable();
+
+    connector.createSegmentTable(segmentsTable);
+    connector.alterSegmentTable();
+    connector.getDBI().withHandle(handle -> {
+      handle.execute("DROP INDEX IDX_B859F28F30657A58518416E445E0AA1A78A68177");
+      handle.execute(StringUtils.format("CREATE INDEX IDX_%1$s_USED ON %1$s(used)", segmentsTable));
+      return null;
+    });
+
+    connector.createSegmentTable(segmentsTable);
+    connector.alterSegmentTable();
+
+    final Set<String> expectedIndices = Sets.newHashSet(
+        StringUtils.format("IDX_%S_USED", segmentsTable),
```

Comment:
Let's use the final index name where possible since we know what they are going to be.

```suggestion
        "IDX_DRUIDTEST_SEGMENTS_USED",
```